### ISSUE TITLE
HttpPort parameter doesn't works

### ISSUE THREAD

**Issue by leoblanc** [helm/charts#1614]
**Is this a request for help?**: No

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**: Using Kubernetes 1.6.6 and Helm 2.5.0

**Which chart**: Consul (https://github.com/kubernetes/charts/tree/master/stable/consul)

**What happened**: After assign the HttpPort parameter to 80, the Consul UI can't be reached (I'm using LoadBalancer as Service Type)

**What you expected to happen**: See the Consul UI (as it's shown when you use the default port 8500)

**How to reproduce it** (as minimally and precisely as possible): Just set the HttpPort to a value different from 8500. For example:

```
   helm install \
     --name consul \
     --namespace default \
     --set "uiService.enabled=true,uiService.type=LoadBalancer,HttpPort=80" \
     stable/consul
```

**Anything else we need to know**: By the way, the "ui.enabled" parameter described on the documentation it's not used at all on the Helm Chart, but that's a different issue.

I have received the help of @netflash to investigate this issue. He discovered that, when "uiService.enabled" is set to true, only the "-ui" argument is passed to the Consul server (https://github.com/kubernetes/charts/blob/master/stable/consul/templates/consul.yaml#L179). This is ok, but is not enough if "HttpPort" was also used to indicate a different port. According to https://www.consul.io/docs/agent/options.html, the "ports/http" setting should be changed also:
![ports-http](https://user-images.githubusercontent.com/25181059/28943241-741c97de-7874-11e7-9450-71116c41fe67.png)

Thank you very much in advance!

**Comment by jonstacks** [comment]
@leoblanc,

Good find. Thanks for the great details. I was able to re-produce from the example you gave. I'm hoping the above PR will fix your issue. Also, just curious, is there any reason you are running the UI Service as a Load Balancer instead of an Ingress? Trying to find out how best this chart can be tailored to the community's needs.

**Comment by leoblanc** [comment]
@jonstacks Hi Jonathan, thank you very much for [your patch](https://github.com/kubernetes/charts/pull/3375/files)! Yes, I think that it solves this issue, hope it can be merged soon. Regarding your question: we found this problem trying to create a minimal prototype of Consul on k8s on AWS, and that's why we used a LoadBalancer. We thought that fix this issue could be useful for other people too, that's why we reported the issue. By the way, we were contacted by a real client recently to deploy Consul on k8s, so I think that we will be using this chart in production soon. Thanks again!
